src/Type/TypehintHelper.php

```diff
@@ -56,7 +56,7 @@ public static function decideTypeFromReflection(
 			return $phpDocType ?? new MixedType();
 		}
 
-		$reflectionTypeString = (string) $reflectionType;
+		$reflectionTypeString = $reflectionType->getName();
```

Comment:
This method only exists on ReflectionNamedType but `$reflectionTypeString` is of type `ReflectionType`.

Reply:
See https://github.com/phpstan/phpstan/issues/1133

Reply:
I see that you replied there in the past. I'd like to know what is the safe way to implement this since `__toString` on ReflectionType is deprecated?

Reply:
At this moment, `ReflectionNamedType` is the only subclass of `ReflectionType` so even an `assert($reflectionTypeString instanceof ReflectionNamedType)` would be fine here IMO, until union/intersection types land.

The purpose of `ReflectionNamedType` is to leave door open for composite types in future (i.e. `ReflectionUnionType`), like union/intersection types where `getName()` wouldn't make sense and this code would need to be adjusted for different `ReflectionType` subtypes.

So this code: https://phpstan.org/r/e837a343-97f8-40a0-a637-fcb56c230927 should produce an error already but it doesn't due to the hack in 9a9b549beaf9b868ed402394010a9b722f47d616 (which isn't true because the method doesn't exist: https://3v4l.org/1M57o).

And since `ReflectionType` is not sealed by PHP (like e.g. `DateTimeInterface`) and can be extended in userland, I think 9a9b549beaf9b868ed402394010a9b722f47d616 is wrong as the assumption wouldn't work properly e.g. with BetterReflection: https://github.com/Roave/BetterReflection/blob/3.5.0/src/Reflection/ReflectionType.php

Reply:
I'd say this is bad design on PHP part that blocks taking advantage of polymorphism. `ReflectionType` should have a method that makes sense in all descendants, like `getTypes` or something.

I'll modify PHP code to always check for `ReflectionNamedType`. But I'm against removing `getType` from `ReflectionType` right now because that issue is still very theoretical. Even PhpStorm's stubs include it.

Reply:
> Even PhpStorm's stubs include it.

https://github.com/JetBrains/phpstorm-stubs/pull/631

Reply:
@ondrejmirtes FWIW the fix has been merged in JetBrains stubs.